crates/bevy_transform/src/components/transform.rs

```diff
@@ -288,6 +288,43 @@ impl Transform {
         self.local_z()
     }
 
+    /// Translates the entity in the local space using a given direction, speed, and delta time.
+    ///
+    /// The translation is applied directly to the entity's local position without considering
+    /// the entity's rotation. This means the entity moves along the provided direction in its
+    /// own local space.
+    ///
+    /// # Parameters
+    /// - `direction`: A `Vec3` representing the direction of movement in the entity's local space.
+    /// - `speed`: A `f32` representing the speed at which the entity should move.
+    /// - `delta_time_seconds`: A `f32` representing the time elapsed in seconds, typically
+    ///   the time since the last frame.
+    pub fn translate(&mut self, direction: Vec3, speed: f32, delta_time_seconds: f32) {
```

Comment:
I think we should merge `direction` and `speed` into a single `velocity: Vec3`, else we're adding redundant ways to specify speed.

Alternatively if we really want to separate direction and speed (which in my opinion is not a necessary requirement since we multiply them back together anyway). It'd be best to use a `Dir3` type for the direction, as it ensures the direction vector is normalised (length 1).

Reply:
I do like your suggestion to be honest.
Didn't think about using the velocity as the result of the multiplication of direction and speed.

Going to implement that straight away!